public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -35,12 +36,19 @@ export class PrometheusDatasource implements DataSourceApi<PromQuery> {
   resultTransformer: ResultTransformer;
 
   /** @ngInject */
-  constructor(instanceSettings, private $q, private backendSrv: BackendSrv, private templateSrv, private timeSrv) {
+  constructor(
+    instanceSettings: DataSourceInstanceSettings<PromOptions>,
+    private $q,
+    private backendSrv: BackendSrv,
+    private templateSrv: TemplateSrv,
+    private timeSrv: TimeSrv
+  ) {
+    super(instanceSettings);
+
     this.type = 'prometheus';
     this.editorSrc = 'app/features/prometheus/partials/query.editor.html';
-    this.name = instanceSettings.name;
     this.url = instanceSettings.url;
-    this.directUrl = instanceSettings.directUrl;
+    this.directUrl = (instanceSettings as any).directUrl; // CHECK!!!
```

Comment:
@torkelo -- I'm not sure what `directUrl` will do here.  my guess is it gets ignored

Reply:
looks like a special Prometheus thing, special handling in the backend that set's the full direct url here (not the proxy url) to link to prometheus in the query editor.

Reply:
But how does it get saved?  IIUC, the backend knows how to handle everythign in instanceSettings and some of the fields in jsonData, 

but what happens if you put other fields on `instanceSettings` -- what does the backend do?  what should it do?

Reply:
It does not get saved, only a different projection set in frontendsettings.go  so that Prometheus has access to both proxy url and direct url

Reply:
so what should we do here?  keep the `any` cast?